src/algorithms/standard/idct.cpp

```diff
@@ -0,0 +1,155 @@
+"- compute MFCC\n"
+"- smoothedMelBands = 10^(IDCT(MFCC)/20) \n"
+"\n"
+"Note: The 'inputSize' parameter is only used as an optimization when the algorithm is configured. "
+"The IDCT will automatically adjust to the size of any input.\n"
+"\n"
+"References:\n"
+"  [1] Discrete cosine transform - Wikipedia, the free encyclopedia,\n"
+"  http://en.wikipedia.org/wiki/Discrete_cosine_transform \n"
+"  [2] HTK book, chapter 5.6 ,\n"
+"  http://speech.ee.ntu.edu.tw/homework/DSP_HW2-1/htkbook.pdf");
+
+
+void IDCT::configure() {
+  int inputSize = parameter("inputSize").toInt();
+  _outputSize = parameter("outputSize").toInt();
+  _type = parameter("dctType").toInt();
+  _lifter = parameter("liftering").toInt();
+  if (_type == 2){
+    createIDctTableII(inputSize, _outputSize);
+  }
+  else if (_type == 3){
+    createIDctTableIII(inputSize, _outputSize);
+  }
+  else {
+    throw EssentiaException("IDCT: Bad DCT type.");
+  }
+}
+
+void IDCT::createIDctTableII(int inputSize, int outputSize) {
+  // simple implementation using matrix multiplication, can probably be sped up
+  // using a library like FFTW, for instance.
+  if (outputSize < inputSize) {
+    throw EssentiaException("IDCT: 'outputSize' is smaller than 'inputSize'. You can only compute the IDCT with an output size greater or equal than the input size");
+  }
+
+  _idctTable = vector<vector<Real> >(outputSize, vector<Real>(inputSize));
+
+  // scale for index = 0
+  Real scale0 = 1.0 / sqrt(Real(outputSize));
+
+  // scale for index != 0
+  Real scale1 = Real(sqrt(2.0/outputSize));
+
+  for (int i=0; i<inputSize; ++i) {
+    Real scale = (i==0)? scale0 : scale1;
+
+    Real freqMultiplier = Real(M_PI / outputSize * i);
+
+    for (int j=0; j<outputSize; ++j) {
+      _idctTable[j][i] = (Real)(scale * cos( freqMultiplier * ((Real)j + 0.5) ));
+    }
+  }
+}
+
+void IDCT::createIDctTableIII(int inputSize, int outputSize) {
+  // simple implementation using matrix multiplication, can probably be sped up
+  // using a library like FFTW, for instance.
+  if (outputSize < inputSize) {
+    throw EssentiaException("IDCT: 'outputSize' is smaller than 'inputSize'. You can only compute the IDCT with an output size greater or equal than the input size");
+  }
+
+  _idctTable = vector<vector<Real> >(outputSize, vector<Real>(inputSize));
+  // This implementation is used instead of the referenced in order to match the behaviour of the HTK
+  // http://speech.ee.ntu.edu.tw/homework/DSP_HW2-1/htkbook.pdf
+
+   Real scaleDefault = Real(sqrt(2.0/outputSize));
+
+   for (int i=0; i<inputSize; ++i) {
+     Real freqMultiplier = Real(M_PI / outputSize * i);
+     Real scale = (i == 0) ? scaleDefault/2 : scaleDefault;
+     for (int j=0; j<outputSize; ++j) {
+       _idctTable[j][i] = (Real)(scale * cos( freqMultiplier * ( (Real)j + 0.5) ) );
+
+     }
+   }
+}
+
+
+void IDCT::compute() {
+
+  const vector<Real>& input = _dct.get();
+  vector<Real>& idct = _idct.get();
+
+  vector<Real> dct = input;
+
```

Comment:
Here I copied the input to a new vector in order to be able to use it. Is there a faster way to do it? @dbogdanov